src/shiver/models/makeslices.py

```diff
@@ -0,0 +1,239 @@
+    IMDHistoWorkspaceProperty,
+)
+
+
+from mantid.kernel import (
+    Direction,
+)
+
+from mantid.simpleapi import (
+    MinusMD,
+    DeleteWorkspaces,
+    Comment,
+    MakeSlice,
+    FlippingRatioCorrectionMD,
+    _create_algorithm_function,
+)
+
+
+from shiver.version import __version__
+
+
+class MakeMultipleSlices(DataProcessorAlgorithm):
+    # pylint: disable=invalid-name,missing-function-docstring
+    """MakeMultipleSlices algorithm"""
+
+    def name(self):
+        return "MakeMultipleSlices"
+
+    def category(self):
+        return "Shiver"
+
+    def PyInit(self):
+        self.declareProperty(
+            IMDEventWorkspaceProperty("SFInputWorkspace", defaultValue="", direction=Direction.Input),
+            doc="InputWorkspace Spin-Flip MDEvent workspace. Must be in Q_sample frame.",
+        )
+
+        self.declareProperty(
+            IMDEventWorkspaceProperty("NSFInputWorkspace", defaultValue="", direction=Direction.Input),
+            doc="InputWorkspace Non Spin-Flip MDEvent workspace. Must be in Q_sample frame.",
+        )
+
+        self.declareProperty(
+            name="FlippingRatio",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="""Formula to define the flipping ratio. It can depend on the variables in the list of sample logs
+          defined below""",
+        )
+
+        self.declareProperty(
+            name="FlippingSampleLog",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="""
+            Comma separated list of sample logs that can appear in the formula for flipping ratio""",
+        )
+
+        self.copyProperties(
+            "MakeSlice",
+            [
+                "BackgroundWorkspace",
+                "NormalizationWorkspace",
+                "QDimension0",
+                "QDimension1",
+                "QDimension2",
+                "Dimension0Name",
+                "Dimension0Binning",
+                "Dimension1Name",
+                "Dimension1Binning",
+                "Dimension2Name",
+                "Dimension2Binning",
+                "Dimension3Name",
+                "Dimension3Binning",
+                "SymmetryOperations",
+                "Smoothing",
+            ],
+        )
+
+        self.declareProperty(
+            IMDHistoWorkspaceProperty(
+                "SFOutputWorkspace", defaultValue="", optional=PropertyMode.Mandatory, direction=Direction.Output
+            ),
+            doc="Spin-Flip OutputWorkspace IMDHisto workspace",
+        )
+        self.declareProperty(
+            IMDHistoWorkspaceProperty(
+                "NSFOutputWorkspace", defaultValue="", optional=PropertyMode.Mandatory, direction=Direction.Output
+            ),
+            doc="Non Spin-Flip OutputWorkspace IMDHisto workspace",
+        )
+
+    def PyExec(self):
+        flipping_ratio = self.getPropertyValue("FlippingRatio")
+
+        if flipping_ratio == "":
```

Comment:
Add a validator instead in the PyInit function

Reply:
done!